File: fix_hooks.js
const fs = require('fs');
let code = fs.readFileSync('thrive/hope-survey.html', 'utf8');

// The block we want to extract starts at:
// "    // --- RESULTS SCREEN ---\n    if (step === 'results') {"
// and ends right before:
// "    // --- SURVEY SCREEN ---"

const resultsStartStr = "    // --- RESULTS SCREEN ---\n    if (step === 'results') {";
const resultsEndStr = "    // --- SURVEY SCREEN ---";

let startIndex = code.indexOf(resultsStartStr);
let endIndex = code.indexOf(resultsEndStr);

if (startIndex === -1 || endIndex === -1) {
    console.error('Could not find results block bounds.');
    process.exit(1);
}

// Get the results block body
let resultsBlock = code.substring(startIndex, endIndex);

// We want to transform the inside of the if statement into a component.
// We remove the `if (step === 'results') {` and the matching closing `}`.
// Let's find the closing `}` which should be just before `// --- SURVEY SCREEN ---`.

let newResultsScreenCode = `
const ResultsScreen = ({ answers, surveyQuestions }) => {
` + resultsBlock.substring(resultsStartStr.length, resultsBlock.lastIndexOf('}')) + `
};
`;

// Now replace the old results block in the App with just a component call
let newAppResultsBlock = `    // --- RESULTS SCREEN ---
    if (step === 'results') {
        return <ResultsScreen answers={answers} surveyQuestions={surveyQuestions} />;
    }

`;

// Put it all together
// 1. Insert ResultsScreen before App
let finalCode = code.replace("const App = () => {", newResultsScreenCode + "\nconst App = () => {");
// 2. Replace the inside of App
finalCode = finalCode.replace(resultsBlock, newAppResultsBlock);

fs.writeFileSync('thrive/hope-survey.html', finalCode);
console.log('Fixed React Hooks error successfully!');

File: fix_hooks.py
import re

with open('thrive/hope-survey.html', 'r') as f:
    code = f.read()

results_start_str = "    // --- RESULTS SCREEN ---\n    if (step === 'results') {"
results_end_str = "    // --- SURVEY SCREEN ---"

start_idx = code.find(results_start_str)
end_idx = code.find(results_end_str)

if start_idx == -1 or end_idx == -1:
    print("Could not find results block bounds.")
    exit(1)

results_block = code[start_idx:end_idx]

# We need to find the closing brace of the if statement.
# The end of results_block is right before // --- SURVEY SCREEN ---
# So it ends with:
# "            </main>\n        );\n    }\n\n"
# Let's extract the inside of the if statement.
inside = results_block[len(results_start_str):results_block.rfind('}')]

new_results_component = """
const ResultsScreen = ({ answers, surveyQuestions }) => {""" + inside + """
};
"""

new_app_results_block = """    // --- RESULTS SCREEN ---
    if (step === 'results') {
        return <ResultsScreen answers={answers} surveyQuestions={surveyQuestions} />;
    }

"""

# Insert ResultsScreen right before App
final_code = code.replace("const App = () => {", new_results_component + "\nconst App = () => {")

# Replace old results block inside App
final_code = final_code.replace(results_block, new_app_results_block)

with open('thrive/hope-survey.html', 'w') as f:
    f.write(final_code)

print("Fixed React Hooks error successfully!")
